ocaml/xapi/xapi_vgpu.ml

```diff
@@ -66,12 +68,13 @@ let atomic_set_resident_on ~__context ~self ~value = assert false
 
 let copy ~__context ~vm vgpu =
   let all = Db.VGPU.get_record ~__context ~self:vgpu in
-  let vgpu = create ~__context
+  let vgpu = create' ~__context
```

Comment:
I would add a comment explaining that here you expect only
```
| Operation                 | expected VM power_state during VGPU.copy |
+---------------------------+------------------------------------------+
| VM.checkpoint             | Suspended                                |
| VM.snapshot               | Halted                                   |
| VM.revert_from_checkpoint | Halted                                   |
| VM.clone                  | Halted                                   |
```

Why not making the expected power state explicit here in copy as well?

Reply:
yes, that would be good, I'll update the comment

Reply:
As robhoes suggested, it's probably simpler to just have a boolean flag to indicate that the powerstate should not be checked during copy, so I think there's no need to check states in copy -- it should just copy regardless of the vm state.

Reply:
True. I agree